aws/resource_aws_transfer_server_test.go

```diff
@@ -111,6 +111,39 @@ func TestAccAWSTransferServer_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSTransferServer_protocols_basic(t *testing.T) {
+	var conf transfer.DescribedServer
+	rName := acctest.RandString(5)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t); testAccPreCheckAWSTransfer(t) },
+		IDRefreshName: "aws_transfer_server.foo",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSTransferServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSTransferServerConfig_protocols_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSTransferServerExists("aws_transfer_server.foo", &conf),
+					testAccMatchResourceAttrRegionalARN("aws_transfer_server.foo", "arn", "transfer", regexp.MustCompile(`server/.+`)),
+					resource.TestMatchResourceAttr(
+						"aws_transfer_server.foo", "endpoint", regexp.MustCompile(fmt.Sprintf("^s-[a-z0-9]+.server.transfer.%s.amazonaws.com$", testAccGetRegion()))),
+					resource.TestCheckResourceAttr(
+						"aws_transfer_server.foo", "identity_provider_type", "SERVICE_MANAGED"),
+					resource.TestCheckResourceAttr("aws_transfer_server.foo", "tags.%", "2"),
+					resource.TestCheckResourceAttr("aws_transfer_server.foo", "protocols.#", "1"),
+				),
```

Comment:
@ewbankkit i'd had wanted to explicitly check the value here like this: 
```hcl
resource.TestCheckResourceAttr("aws_transfer_server.foo", "protocols.[HASH]", transfer.ProtocolSftp),

```
using the hash of the property but unable to find documentation on how to do that.

Reply:
This can be done via

```go
import "github.com/terraform-providers/terraform-provider-aws/aws/internal/tfawsresource"

...

tfawsresource.TestCheckTypeSetElemAttr("aws_transfer_server.foo", "protocols.*", transfer.ProtocolSftp),
```